packages/react-components/lib/doors/door-table-datagrid.tsx

```diff
@@ -0,0 +1,152 @@
+import { DataGrid, GridColDef, GridValueGetterParams, GridCellParams } from '@mui/x-data-grid';
+import { Box, Button, SxProps, Typography, useTheme } from '@mui/material';
+import * as React from 'react';
+import { DoorState } from 'api-client';
+import { DoorMode } from 'rmf-models';
+import { doorModeToString, doorTypeToString } from './utils';
+
+export interface DoorTableData {
+  index: number;
+  doorName: string;
+  levelName: string;
+  doorType: number;
+  doorState?: DoorState;
+  onClickOpen?(): void;
+  onClickClose?(): void;
+}
+
+export interface DoorDataGridTableProps {
+  doors: DoorTableData[];
+}
+
+export function DoorDataGridTable({ doors }: DoorDataGridTableProps): JSX.Element {
+  const theme = useTheme();
+
+  const DoorState = (params: GridCellParams): React.ReactNode => {
+    const labelStyle: SxProps = React.useMemo<SxProps>(() => {
+      const disabled = {
+        color: theme.palette.action.disabledBackground,
+      };
+      const open = {
+        color: theme.palette.success.main,
+      };
+      const closed = {
+        color: theme.palette.error.main,
+      };
+      const moving = {
+        color: theme.palette.warning.main,
+      };
+
+      switch (params.row.doorState.current_mode.value) {
+        case DoorMode.MODE_OPEN:
+          return open;
+        case DoorMode.MODE_CLOSED:
+          return closed;
+        case DoorMode.MODE_MOVING:
+          return moving;
+        default:
+          return disabled;
+      }
+    }, [params.row.doorState.current_mode.value]);
+
+    return (
+      <Box sx={labelStyle}>
+        <Typography
+          data-testid="door-state"
+          component="p"
+          sx={{
+            fontWeight: 'bold',
+            fontSize: 14,
+          }}
+        >
+          {params.row.doorState ? doorModeToString(params.row.doorState.current_mode.value) : -1}
+        </Typography>
+      </Box>
+    );
+  };
+
+  const OpenCloseButtons = (params: GridCellParams): React.ReactNode => {
+    return (
+      <Box sx={{ margin: 0, padding: 0, paddingRight: 1 }}>
+        <Button
+          variant="contained"
+          size="small"
+          aria-label="open"
+          sx={{ marginRight: 2 }}
+          onClick={params.row.onClickOpen}
+        >
+          Open
+        </Button>
+        <Button
+          variant="contained"
+          size="small"
+          aria-label="close"
+          onClick={params.row.onClickClose}
+        >
+          Close
+        </Button>
+      </Box>
+    );
+  };
+
+  const columns: GridColDef[] = [
+    {
+      field: 'doorName',
+      headerName: 'Name',
+      width: 90,
+      valueGetter: (params: GridValueGetterParams) => params.row.doorName,
+      flex: 1,
+      filterable: true,
+    },
+    {
+      field: 'levelName',
+      headerName: 'Current floor',
```

Comment:
```suggestion
      headerName: 'Current Floor',
```
nit: capitalize second words similar figma and how we did for robot table and task table. This comment applies to the other columns and their tests as well then.